spec/helpers/faker/puppet_module_remote_repo.rb

```diff
@@ -0,0 +1,130 @@
+require 'open3'
+
+require_relative '../faker'
+
+# Fake a remote git repository that hold a puppet module
+#
+# This module allow to fake a remote repositiory using:
+#  - a bare repo
+#  - a temporary cloned repo to operate on the remote before exposition to modulesync
+#
+# Note: This module need to have working_directory sets before using it
+module Faker
+  class PuppetModuleRemoteRepo
+    attr_reader :name
+    attr_reader :namespace
+
+    def initialize(name, namespace)
+      @name = name
+      @namespace = namespace
+    end
+
+    def populate
+      FileUtils.chdir(Faker.working_directory) do
+        run "git init --bare '#{bare_repo_dir}'"
+        run "git clone '#{bare_repo_dir}' '#{tmp_repo_dir}'"
+
+        module_short_name = name.sub(/^puppet-/, '')
+
+        FileUtils.chdir(tmp_repo_dir) do
+          metadata = {
+            name: "modulesync-#{module_short_name}",
+            version: '0.4.2',
+            author: 'ModuleSync team',
+          }.to_json
+
+          File.write 'metadata.json', metadata
+          run 'git add metadata.json'
+          run "git commit -m'Initial commit'"
+          run 'git push'
+        end
+      end
+    end
+
+    def read_only=(value)
+      mode = value ? '0444' : '0644'
+      FileUtils.chdir(bare_repo_dir) do
+        run "git config core.sharedRepository #{mode}"
+      end
+    end
+
+    def default_branch=(value)
+      FileUtils.chdir(bare_repo_dir) do
+        run "git branch -M master #{value}"
+        run "git symbolic-ref HEAD refs/heads/#{value}"
+      end
+    end
```

Comment:
Git 2.28 has `init.defaultBranch` as a config item. Perhaps the code should have a comment that mentions this. I assume you went for this route to be compatible since 2.28 is still very new.